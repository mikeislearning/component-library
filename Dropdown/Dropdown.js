import React from 'react';
import { Image, View } from 'react-native';
import { PropTypes } from 'prop-types';
import RNPickerSelect from 'react-native-picker-select';
import { images } from 'app/theme';
import { FormError } from 'app/components';

import styles from './Dropdown.styles';

export default class Dropdown extends React.Component {
  static propTypes = {
    errorMsg: PropTypes.string,
    errors: PropTypes.bool,
    value: PropTypes.string,
    placeholder: PropTypes.string,
    onValueChange: PropTypes.func,
    onUpArrow: PropTypes.func,
    onDownArrow: PropTypes.func,
    dropdownRef: PropTypes.func,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.string,
      })
    ),
  };

  render() {
    const {
      errorMsg,
      errors,
      value,
      placeholder,
      items,
      onValueChange,
      onUpArrow,
      onDownArrow,
      dropdownRef,
    } = this.props;

    const formatedPlaceholder = {
      label: placeholder,
      value: null,
    };

    if (!items.length) {
      return null;
    }

    let inputStyles = styles.input;
    if (!value) {
      inputStyles = { ...inputStyles, ...styles.inputEmpty };
    }
    if (errors) {
      inputStyles = { ...inputStyles, ...styles.errorInput };
    }

    return (
      <View style={styles.inputContainer}>
        <RNPickerSelect
          Icon={() => {
            return <Image source={images.arrowDropdown} style={styles.arrowDropdown} />;
          }}
          items={items}
          onValueChange={onValueChange}
          onUpArrow={onUpArrow}
          onDownArrow={onDownArrow}
          placeholder={formatedPlaceholder}
          style={{
            inputIOS: inputStyles,
            inputAndroid: inputStyles,
            chevronActive: styles.chevronActive,
            placeholder: styles.placeholder,
          }}
          useNativeAndroidPickerStyle={false}
          value={value}
          ref={dropdownRef}
        />
        {errorMsg && <FormError text={errorMsg} />}
      </View>
    );
  }
}
