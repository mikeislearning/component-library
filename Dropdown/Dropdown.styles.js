import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  arrowDropdown: {
    height: 8,
    right: 10,
    top: 22,
    width: 8,
  },
  errorInput: {
    borderColor: colors.red.light,
  },
  input: {
    backgroundColor: colors.white,
    borderColor: colors.black,
    borderRadius: 8,
    borderWidth: 1,
    color: colors.black,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
    height: 50,
    minWidth: 130,
    paddingHorizontal: 10,
  },
  inputContainer: {
    marginBottom: 10,
  },
  inputEmpty: {
    backgroundColor: colors.gray.light,
    borderColor: colors.gray.medium,
    fontFamily: fonts.family.messinaSans.lightItalic,
  },
  placeholder: {
    color: colors.gray.medium,
  },
});

export default styles;
