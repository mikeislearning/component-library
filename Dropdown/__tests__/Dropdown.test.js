import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Dropdown from '../Dropdown';

const itemList = [
  {
    label: 'Label1',
    value: 'Value1',
  },
  {
    label: 'Label2',
    value: 'Value2',
  },
  {
    label: 'Label3',
    value: 'Value3',
  },
];

describe('<Dropdown />', () => {
  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Dropdown placeholder="Placeholder text" items={itemList} onValueChange={() => {}} />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
