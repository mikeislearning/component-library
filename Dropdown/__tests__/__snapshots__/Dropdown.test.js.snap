// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly 1`] = `
<React.Fragment>
  <RNPickerSelect
    Icon={[Function]}
    InputAccessoryView={null}
    disabled={false}
    doneText="Done"
    hideDoneBar={false}
    itemKey={null}
    items={
      Array [
        Object {
          "label": "Label1",
          "value": "Value1",
        },
        Object {
          "label": "Label2",
          "value": "Value2",
        },
        Object {
          "label": "Label3",
          "value": "Value3",
        },
      ]
    }
    modalProps={Object {}}
    onClose={null}
    onDonePress={null}
    onDownArrow={null}
    onOpen={null}
    onUpArrow={null}
    onValueChange={[Function]}
    pickerProps={Object {}}
    placeholder={
      Object {
        "label": "Placeholder text",
        "value": null,
      }
    }
    placeholderTextColor="#C7C7CD"
    style={
      Object {
        "chevronActive": undefined,
        "inputAndroid": Object {
          "backgroundColor": "#ECECEC",
          "borderColor": "#726E67",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#000000",
          "fontFamily": "MessinaSans-LightItalic",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 10,
          "minWidth": 130,
          "paddingHorizontal": 10,
        },
        "inputIOS": Object {
          "backgroundColor": "#ECECEC",
          "borderColor": "#726E67",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "#000000",
          "fontFamily": "MessinaSans-LightItalic",
          "fontSize": 16,
          "height": 50,
          "marginBottom": 10,
          "minWidth": 130,
          "paddingHorizontal": 10,
        },
        "placeholder": Object {
          "color": "#726E67",
        },
      }
    }
    textInputProps={Object {}}
    useNativeAndroidPickerStyle={false}
  />
</React.Fragment>
`;
